icu4j/main/core/src/main/java/com/ibm/icu/impl/LocaleIDParser.java

```diff
@@ -432,10 +441,15 @@ private int parseVariant() {
                     c = UNDERSCORE;
                 }
                 append(c);
+                if (buffer.length() - oldBlen > MAX_VARIANTS_LENGTH) {
+                    throw new IllegalArgumentException("variants is too long");
+                }
             }
         }
         --index; // unget
-
+        if (buffer.length() - oldBlen > MAX_VARIANTS_LENGTH) {
```

Comment:
It seems to me that this could never happen. Are there any circumstances under which this code could actually be reached? If not, it won't ever be used and can be deleted.

Reply:
yes this is correct. I removed it.